app/assets/javascripts/qpixel_api.js

```diff
@@ -177,29 +179,28 @@ window.QPixel = {
    * @returns {Promise<Object>} a JSON object containing user preferences
    */
   preferences: async () => {
-    if (this._preferences == null && !!localStorage['qpixel.user_preferences']) {
-      this._preferences = JSON.parse(localStorage['qpixel.user_preferences']);
-
-      // If we don't have the global key, we're probably using an old preferences schema.
-      if (!this._preferences.global) {
-        delete localStorage['qpixel.user_preferences'];
-        this._preferences = null;
-      }
+    // Do not attempt to access preferences if user is not signed in
+    const user = await QPixel.user();
+    if ('error' in user) {
```

Comment:
consider:

```suggestion
    if (!!user.error) {
```

This is entirely opinion, but I find the `'property' in object` form confusing - on the face of it, it looks like a string operation of some sort, when it's not. The above is an alternative.

Reply:
I know the syntax of `'error' in user` is a bit odd (especially coming from Python, where `in` can be used for checking for substrings, and where I can write `'error' in user.keys()` to make it clearer what's being checked).

I still prefer `'error' in user` over the cast to Boolean for the avoidance of ambiguity:
- `if('error' in user)` is only false if `user` does not contain a key called `'error'`
- `if(!!user.error)` is false if either `user` does not contain a key called `'error'`, or `user` does contain a key called `'error'` whose corresponding value is anything falsy

In this particular case it's unlikely `user.error` would ever contain something falsy, but in general that's my reason for preferring the explicit key check.

---

Just checked out of curiosity and it is possible to be even more explicit with JavaScript, similarly to Python but a little more verbose:

`if (Object.keys(user).includes('error'))`

I don't feel strongly enough about this particular case to argue against any of the mentioned forms though.